### ISSUE TITLE
Crash on 0.26

### ISSUE THREAD

**Issue by levone1** [vamolessa/pepper#43]
Just installed 0.26, and get 
`thread 'main' panicked at 'could not spawn server', /home/defaultuser/.cargo/registry/src/github.com-1ecc6299db9ec823/pepper-0.26.0/src/platforms/unix_utils.rs:87:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`

Using SailfishOS (mobile).
Previous version worked fine...

**Comment by vamolessa** [comment]
Wait, were you using pepper on a mobile device running SailfishOS?
How was it invoked? Were you using a shell?

Did you build the binary yourself? If yes, would you try again with the latest commit on `master`?

**Comment by kevinsjoberg** [comment]
I get the same thing on `0.26.0`. Here's the stack trace.

```console
~/code/pepper (master) $ RUST_BACKTRACE=full pepper
~/code/pepper (master) $ RUST_BACKTRACE=full pepper
thread 'main' panicked at 'could not spawn server', /Users/kevin/.asdf/installs/rust/1.60.0/registry/src/github.com-1ecc6299db9ec823/pepper-0.26.0/src/platforms/unix_utils.rs:87:13
stack backtrace:
   0:        0x100aec2d8 - std::backtrace_rs::backtrace::libunwind::trace::h449592924b3bd63f
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:        0x100aec2d8 - std::backtrace_rs::backtrace::trace_unsynchronized::ha2aaeafed0c31c90
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:        0x100aec2d8 - std::sys_common::backtrace::_print_fmt::h58db85a17304976f
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:66:5
   3:        0x100aec2d8 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h10cf06316d33e2a9
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:45:22
   4:        0x100b09bf0 - core::fmt::write::h1faf18c959c3a8df
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/fmt/mod.rs:1190:17
   5:        0x100ae80cc - std::io::Write::write_fmt::h86ab231360bc97d2
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/io/mod.rs:1657:15
   6:        0x100aee5b0 - std::sys_common::backtrace::_print::h771b4aab9b128422
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:48:5
   7:        0x100aee5b0 - std::sys_common::backtrace::print::h637de99a9f76e8a7
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:35:9
   8:        0x100aee5b0 - std::panicking::default_hook::{{closure}}::h36e628ffaf3cd44f
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:295:22
   9:        0x100aee228 - std::panicking::default_hook::h3ee1564a7544e58f
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:314:9
  10:        0x100aba514 - pepper::init::{{closure}}::h0e1d4da099f7aacf
  11:        0x100aeec44 - std::panicking::rust_panic_with_hook::h191339fbd2fe2360
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:702:17
  12:        0x100aee960 - std::panicking::begin_panic_handler::{{closure}}::h91c230befd9929e3
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:586:13
  13:        0x100aec7c0 - std::sys_common::backtrace::__rust_end_short_backtrace::haaaeebb1d37476b3
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/sys_common/backtrace.rs:138:18
  14:        0x100aee6d4 - rust_begin_unwind
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:584:5
  15:        0x100b13ae0 - core::panicking::panic_fmt::h4fe1013b011ef602
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/panicking.rs:143:14
  16:        0x100a92974 - pepper::platform_impl::sys::unix_utils::run::hb24799a8ed5ce225
  17:        0x100aba5dc - pepper::run::h004a3efb196b5b0e
  18:        0x100a5a5a4 - pepper::main::hbc0c8048ef28eb37
  19:        0x100a5a314 - std::sys_common::backtrace::__rust_begin_short_backtrace::h941eae21557ea173
  20:        0x100a5a34c - std::rt::lang_start::{{closure}}::hec6c8f235760203c
  21:        0x100aebb84 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h8eb3ac20f80eabfa
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/core/src/ops/function.rs:259:13
  22:        0x100aebb84 - std::panicking::try::do_call::ha6ddf2c638427188
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:492:40
  23:        0x100aebb84 - std::panicking::try::hda8741de507c1ad0
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:456:19
  24:        0x100aebb84 - std::panic::catch_unwind::h82424a01f258bd39
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panic.rs:137:14
  25:        0x100aebb84 - std::rt::lang_start_internal::{{closure}}::h67e296ed5b030b7b
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/rt.rs:128:48
  26:        0x100aebb84 - std::panicking::try::do_call::hd3dd7e7e10f6424e
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:492:40
  27:        0x100aebb84 - std::panicking::try::ha0a7bd8122e3fb7c
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panicking.rs:456:19
  28:        0x100aebb84 - std::panic::catch_unwind::h809b0e1092e9475d
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/panic.rs:137:14
  29:        0x100aebb84 - std::rt::lang_start_internal::h358b6d58e23c88c7
                               at /rustc/7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c/library/std/src/rt.rs:128:20
  30:        0x100a5a5fc - _main
  ```

**Comment by vamolessa** [comment]
How did you run pepper? Just calling `pepper`?
Do you still get the crash when running `pepper --server` and the running a client (just `pepper`)?

If you built it yourself, would you mind trying the latest commit on master? It adds a little bit of extra info on the panic message.

Thanks!

**Comment by levone1** [comment]
> Wait, were you using pepper on a mobile device running SailfishOS? How was it invoked? Were you using a shell?
> 
> Did you build the binary yourself? If yes, would you try again with the latest commit on `master`? I've just added a little bit of info that may help debug this issue.

Yes, was working fine on SF mobile with 0.24. I'm pretty sure that I installed with `cargo install`. Worked fine invoking with just `pepper`, and with `pepper [file]`. Didn't try with `...server`.

Will build and try again ...

**Comment by levone1** [comment]
0.26.1...
![Screenshot_20220422_009](https://user-images.githubusercontent.com/43003405/164865042-3dcf81d2-e1e2-42b9-84d5-50d4f8ccfb82.png)

**Comment by vamolessa** [comment]
Alright! I can reproduce the crash.
Will work on a fix! Thanks for the reports

**Comment by vamolessa** [comment]
I believe it's now fixed on the released `v0.26.1` build.
Please feel free to reopen if you're still having trouble!
